src/xcode/ENA/ENA/Source/Client/Security/CoronaWarnURLSessionDelegate.swift

```diff
@@ -0,0 +1,143 @@
+//
+// Corona-Warn-App
+//
+// SAP SE and all other contributors /
+// copyright owners license this file to you under the Apache
+// License, Version 2.0 (the "License"); you may not use this
+// file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+//
+
+import Foundation
+import CommonCrypto
+import CryptoKit
+
+final class CoronaWarnURLSessionDelegate: NSObject {
+	// MARK: Known Public Key Storage
+	/// A dictionary containing a mapping of the host to the SHA256 public key string
+	private let domainPublicKeyHashes: [String: String]
+	/// Whitelist of domains we do not pin the public key for. Currently just the distribution URL
+	private let whitelist: [String]
+
+	// MARK: Creating a Delegate
+	override init() {
+		guard
+			let publicKeyDict = Bundle.main.readPlistDict(name: "PublicKeys"),
+			let hostWhitelist = Bundle.main.readPlistAsArr(name: "HostWhitelist")
+		else {
+			preconditionFailure("Could not load PublicKeys.plist for public key pinning!")
+		}
+
+		domainPublicKeyHashes = publicKeyDict
+		whitelist = hostWhitelist
+	}
+}
+
+extension CoronaWarnURLSessionDelegate: URLSessionDelegate {
+	func urlSession(
+		_ session: URLSession,
+		didReceive challenge: URLAuthenticationChallenge,
+		completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void
+	) {
+		func reject() { completionHandler(.cancelAuthenticationChallenge, /* credential */ nil) }
+
+		guard !checkWhitelist(for: challenge.protectionSpace.host) else {
```

Comment:
I would like to cross-check if the correct behaviour is implemented using this whitelist:
1) HTTP requests to all domains != the submission, verification and distribution service domains are not trusted
2) for submission and verification URL's certificate pinning is in place
3) for distribution service domain no certificate pinning is in place

Reply:
Yes, for example a request to google.com will be denied.